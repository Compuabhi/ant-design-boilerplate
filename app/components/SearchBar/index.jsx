import React from 'react';
import { Form, Input, Icon } from 'antd';
import { PropTypes } from 'prop-types';

const FormItem = Form.Item;

require('./styles.css');

const SearchBar = ({ searchText, onChangeSearchText }) => (
  <div className="searchBar">
    <Form layout="inline">
      <FormItem label="Name">
        <Input
          placeholder="Filter by name"
          prefix={<Icon type="search" style={{ color: 'rgba(0,0,0,.25)' }} />}
          value={searchText}
          suffix={searchText ?
            <Icon
              type="close-circle"
              id="search-text-delete"
              onClick={() => onChangeSearchText('')}
            /> : null}
          onChange={e => onChangeSearchText(e.target.value)}
        />
      </FormItem>
    </Form>
  </div>
);

SearchBar.propTypes = {
  searchText: PropTypes.string,
  onChangeSearchText: PropTypes.func,
};

SearchBar.defaultProps = {
  searchText: '',
  onChangeSearchText: () => {},
};

export default SearchBar;
