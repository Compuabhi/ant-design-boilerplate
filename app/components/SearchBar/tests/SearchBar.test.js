import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow, mount } from 'enzyme';
import SearchBar from '../index';

Enzyme.configure({ adapter: new Adapter() });

describe('SearchBar', () => {
  const testProps = {
    searchText: 'test',
    onChangeSearchText: jest.fn(),
  };
  it('should be mounted with no props', () => {
    const wrapper = renderer.create(<SearchBar />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should manage typing with only searchText ', () => {
    const wrapper = shallow(<SearchBar searchText="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with all props', () => {
    const wrapper = renderer.create(<SearchBar {...testProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should dispatch action and react to typing ', () => {
    const wrapper = mount(<SearchBar onChangeSearchText={testProps.onChangeSearchText} />);
    const expected = '111';
    wrapper.find('.ant-input').simulate('change', { target: { value: expected } });
    expect(testProps.onChangeSearchText).toHaveBeenCalledTimes(1);
  });

  it('should delete text when button clear clicked ', () => {
    const wrapper = mount(<SearchBar {...testProps} />);
    wrapper.find('#search-text-delete').at(0).simulate('click');
    expect(testProps.onChangeSearchText).toHaveBeenCalledTimes(2);
  });
});

