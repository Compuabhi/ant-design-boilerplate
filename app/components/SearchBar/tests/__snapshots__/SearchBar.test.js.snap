// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should be mounted with no props 1`] = `
<div
  className="searchBar"
>
  <form
    className="ant-form ant-form-inline"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={undefined}
    >
      <div
        className="ant-form-item-label"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Name"
        >
          Name
        </label>
      </div>
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-affix-wrapper"
              style={undefined}
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  className="anticon anticon-search"
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              </span>
              <input
                className="ant-input"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Filter by name"
                style={null}
                type="text"
                value=""
              />
            </span>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`SearchBar should manage typing with only searchText  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    onChangeSearchText={[Function]}
    searchText="test"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Form
        hideRequiredMark={false}
        layout="inline"
        onSubmit={[Function]}
        prefixCls="ant-form"
      >
        <FormItem
          colon={true}
          hasFeedback={false}
          label="Name"
          prefixCls="ant-form"
        >
          <Input
            disabled={false}
            onChange={[Function]}
            placeholder="Filter by name"
            prefix={
              <Icon
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
                type="search"
              />
            }
            prefixCls="ant-input"
            suffix={
              <Icon
                id="search-text-delete"
                onClick={[Function]}
                type="close-circle"
              />
            }
            type="text"
            value="test"
          />
        </FormItem>
      </Form>,
      "className": "searchBar",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FormItem
          colon={true}
          hasFeedback={false}
          label="Name"
          prefixCls="ant-form"
        >
          <Input
            disabled={false}
            onChange={[Function]}
            placeholder="Filter by name"
            prefix={
              <Icon
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
                type="search"
              />
            }
            prefixCls="ant-input"
            suffix={
              <Icon
                id="search-text-delete"
                onClick={[Function]}
                type="close-circle"
              />
            }
            type="text"
            value="test"
          />
        </FormItem>,
        "hideRequiredMark": false,
        "layout": "inline",
        "onSubmit": [Function],
        "prefixCls": "ant-form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Input
            disabled={false}
            onChange={[Function]}
            placeholder="Filter by name"
            prefix={
              <Icon
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
                type="search"
              />
            }
            prefixCls="ant-input"
            suffix={
              <Icon
                id="search-text-delete"
                onClick={[Function]}
                type="close-circle"
              />
            }
            type="text"
            value="test"
          />,
          "colon": true,
          "hasFeedback": false,
          "label": "Name",
          "prefixCls": "ant-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "onChange": [Function],
            "placeholder": "Filter by name",
            "prefix": <Icon
              style={
                Object {
                  "color": "rgba(0,0,0,.25)",
                }
              }
              type="search"
            />,
            "prefixCls": "ant-input",
            "suffix": <Icon
              id="search-text-delete"
              onClick={[Function]}
              type="close-circle"
            />,
            "type": "text",
            "value": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Form
          hideRequiredMark={false}
          layout="inline"
          onSubmit={[Function]}
          prefixCls="ant-form"
        >
          <FormItem
            colon={true}
            hasFeedback={false}
            label="Name"
            prefixCls="ant-form"
          >
            <Input
              disabled={false}
              onChange={[Function]}
              placeholder="Filter by name"
              prefix={
                <Icon
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                  type="search"
                />
              }
              prefixCls="ant-input"
              suffix={
                <Icon
                  id="search-text-delete"
                  onClick={[Function]}
                  type="close-circle"
                />
              }
              type="text"
              value="test"
            />
          </FormItem>
        </Form>,
        "className": "searchBar",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FormItem
            colon={true}
            hasFeedback={false}
            label="Name"
            prefixCls="ant-form"
          >
            <Input
              disabled={false}
              onChange={[Function]}
              placeholder="Filter by name"
              prefix={
                <Icon
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                  type="search"
                />
              }
              prefixCls="ant-input"
              suffix={
                <Icon
                  id="search-text-delete"
                  onClick={[Function]}
                  type="close-circle"
                />
              }
              type="text"
              value="test"
            />
          </FormItem>,
          "hideRequiredMark": false,
          "layout": "inline",
          "onSubmit": [Function],
          "prefixCls": "ant-form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Input
              disabled={false}
              onChange={[Function]}
              placeholder="Filter by name"
              prefix={
                <Icon
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                  type="search"
                />
              }
              prefixCls="ant-input"
              suffix={
                <Icon
                  id="search-text-delete"
                  onClick={[Function]}
                  type="close-circle"
                />
              }
              type="text"
              value="test"
            />,
            "colon": true,
            "hasFeedback": false,
            "label": "Name",
            "prefixCls": "ant-form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "onChange": [Function],
              "placeholder": "Filter by name",
              "prefix": <Icon
                style={
                  Object {
                    "color": "rgba(0,0,0,.25)",
                  }
                }
                type="search"
              />,
              "prefixCls": "ant-input",
              "suffix": <Icon
                id="search-text-delete"
                onClick={[Function]}
                type="close-circle"
              />,
              "type": "text",
              "value": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`SearchBar should render with all props 1`] = `
<div
  className="searchBar"
>
  <form
    className="ant-form ant-form-inline"
    onSubmit={[Function]}
  >
    <div
      className="ant-row ant-form-item"
      style={undefined}
    >
      <div
        className="ant-form-item-label"
      >
        <label
          className=""
          htmlFor={null}
          onClick={[Function]}
          title="Name"
        >
          Name
        </label>
      </div>
      <div
        className="ant-form-item-control-wrapper"
      >
        <div
          className="ant-form-item-control"
        >
          <span
            className="ant-form-item-children"
          >
            <span
              className="ant-input-affix-wrapper"
              style={undefined}
            >
              <span
                className="ant-input-prefix"
              >
                <i
                  className="anticon anticon-search"
                  style={
                    Object {
                      "color": "rgba(0,0,0,.25)",
                    }
                  }
                />
              </span>
              <input
                className="ant-input"
                disabled={false}
                onChange={[Function]}
                onKeyDown={[Function]}
                placeholder="Filter by name"
                style={null}
                type="text"
                value="test"
              />
              <span
                className="ant-input-suffix"
              >
                <i
                  className="anticon anticon-close-circle"
                  id="search-text-delete"
                  onClick={[Function]}
                />
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
  </form>
</div>
`;
